### DISCUSSION TITLE
tracker updating later than I expected

### DISCUSSION THREAD

**Discussion by bwduncan** [BottlecapDave/HomeAssistant-OctopusEnergy#734] [Q&A]
Hi everyone,

Thanks Dave for an amazing integration :smile: 

I'm using the https://github.com/lozzd/octopus-energy-rates-card which makes it very easy to see the trackers I have configured. (Thanks Laurie!) However, the trackers seem not to be updating as soon as I thought they would. The documentation says that they should update as soon as the data are available, so 16:00. However, today I had to wait until 23:00 to see tomorrow's chosen times. Is this right?

They seem to be updating immediately (within one minute, as expected) now, but this wasn't happening earlier. I even tried deleting and recreating, and creating new trackers, and they did not update until 23:00 tonight, regardless of start times.

I get a feeling this was broken by the latest update, but I can't see anything in the diffs which supports that hunch. If someone thinks it's possible, I'd be happy to downgrade to test it. I asked a friend who is using a (much) older version and he didn't have this problem today...

Thanks!
Bruce

```json
      {
        "entry_id": "f37c8ea80f8620289a96240301e29211",
        "version": 3,
        "minor_version": 1,
        "domain": "octopus_energy",
        "title": "1hr (target)",
        "data": {
          "name": "1hr",
          "hours": 1.0,
          "type": "Continuous",
          "mpan": "123",
          "rolling_target": false,
          "last_rates": false,
          "target_invert_target_rates": false,
          "start_time": "23:00",
          "end_time": "23:00",
          "kind": "target_rate",
          "account_id": "A-456"
        },
        "options": {},
        "pref_disable_new_entities": false,
        "pref_disable_polling": false,
        "source": "user",
        "unique_id": null,
        "disabled_by": null
      },
      {
        "entry_id": "03c1cd3109469db83220ca3653d590ce",
        "version": 3,
        "minor_version": 1,
        "domain": "octopus_energy",
        "title": "1hr_daytime (target)",
        "data": {
          "name": "1hr_daytime",
          "hours": 1.0,
          "type": "Continuous",
          "mpan": "123",
          "rolling_target": false,
          "last_rates": false,
          "target_invert_target_rates": false,
          "start_time": "9:00",
          "end_time": "23:00",
          "kind": "target_rate",
          "account_id": "A-456"
        },
        "options": {
          "name": "1hr_daytime",
          "hours": "1.0",
          "type": "Continuous",
          "mpan": "123",
          "rolling_target": false,
          "last_rates": false,
          "target_invert_target_rates": false,
          "start_time": "9:00",
          "end_time": "16:00",
          "kind": "target_rate",
          "account_id": "A-456"
        },
        "pref_disable_new_entities": false,
        "pref_disable_polling": false,
        "source": "user",
        "unique_id": null,
        "disabled_by": null
      },
      {
        "entry_id": "8c506726c8a35d9d29130bffba24faaa",
        "version": 3,
        "minor_version": 1,
        "domain": "octopus_energy",
        "title": "4hr (target)",
        "data": {
          "name": "4hr",
          "hours": 4.0,
          "type": "Continuous",
          "mpan": "123",
          "rolling_target": false,
          "last_rates": false,
          "target_invert_target_rates": false,
          "start_time": "23:00",
          "end_time": "23:00",
          "kind": "target_rate",
          "account_id": "A-456"
        },
        "options": {},
        "pref_disable_new_entities": false,
        "pref_disable_polling": false,
        "source": "user",
        "unique_id": null,
        "disabled_by": null
      }
```

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/469548/be366931-eeb7-4a58-8fae-25eb3c9a6f49)

**Comment by BottlecapDave** [comment]
Sorry for the late response. This is by design. The target rate sensor will only re-evaluate when all rate data information is available and if either a) the target rates are in the past and rolling target is true or b) the current time period has shifted and rolling target is false. The main reason for this behaviour is to better support debugging of target rate sensors so people can work out why periods were picked and if the sensor doesn't go on, they can see why.

So because the target rate sensor is from 11pm-11pm, this is why it hasn't re-evaluated until 11pm and why your other sensor re-evaluated at 4pm.

**Reply by BottlecapDave** [reply]
In addition there are days when the agile data isn't available bang on 4pm, which can also cause the delay in evaluation.
